pkg/applicationserver/applicationserver.go

```diff
@@ -753,16 +754,16 @@ func (as *ApplicationServer) downlinkQueueOp(ctx context.Context, ids *ttnpb.End
 				}
 			}
 
-			registerReceiveDownlinks(ctx, ids, items)
+			registerReceiveDownlinks(ctx, dev.Ids, items)
```

Comment:
The registry always contains the full set of identifiers for an end device, while the caller provided `ids` may be only minimalistic (application ID + end device ID).